hail/src/main/scala/is/hail/utils/PartitionCounts.scala

```diff
@@ -0,0 +1,82 @@
+package is.hail.utils
+
+object PartitionCounts {
+
+  def getPCSubsetOffset(n: Long, pcs: Iterator[Long]): Option[(Int, Long, Long)] = {
+    var i = 0
+    var nLeft = n
+    for (c <- pcs) {
+      if (c >= nLeft)
+        return Some((i, nLeft, c - nLeft))
+      i = i + 1
+      nLeft = nLeft - c
+    }
+    return None
+  }
+
+  def getHeadPCs(original: IndexedSeq[Long], n: Long): IndexedSeq[Long] =
+    getPCSubsetOffset(n, original.iterator) match {
+      case Some((lastIdx, nKeep, nDrop)) =>
+        (0 to lastIdx).map { i =>
+          if (i == lastIdx)
+            nKeep
+          else
+            original(i)
+        }
+      case None => original
+    }
+
+  def getTailPCs(original: IndexedSeq[Long], n: Long): IndexedSeq[Long] =
+    getPCSubsetOffset(n, original.reverseIterator) match {
+      case Some((lastIdx, nKeep, nDrop)) =>
+        (0 to lastIdx).reverseMap { i =>
+          if (i == lastIdx)
+            nKeep
+          else
+            original(original.length - i - 1)
+        }
+      case None => original
+    }
+
+  def incrementalPCSubsetOffset(
+    n: Long,
+    partIndices: IndexedSeq[Int]
+  )(computePCs: IndexedSeq[Int] => IndexedSeq[Long]): (Int, Long, Long) = { // (finalIndex, nKeep, nDrop)
```

Comment:
Instead of a comment, let's make a case class:

```scala
case class PCSubsetOffset(
  val finalIndex: Int,
  val nKeep: Long,
  val nDrop: Long
)
```
The match statements above will get a bit more verbose, but I think it's worth the self-documenting clarity elsewhere.